projects/plugins/jetpack/_inc/client/sharing/publicize.jsx

```diff
@@ -154,10 +155,14 @@ export const Publicize = withModuleSettingsFormHelpers(
 							>
 								{ __( 'Automatically share your posts to social networks', 'jetpack' ) }
 							</ModuleToggle>
-							{ shouldShowChildElements && hasAutoConversion && <AutoConversionSection /> }
-							{ shouldShowChildElements && hasSocialImageGenerator && (
-								<SocialImageGeneratorSection />
-							) }
+							<RefreshJetpackSocialSettingsWrapper
```

Comment:
Are we using this component here so we can encapsulate the `useEffect` because this is a class component?

That's probably fine, but it might be worth checking that we don't want to do something like handle it with a lifecycle method in here.

Reply:
`useEffect` is not even needed for this and thus a lifecycle method may be a good place for it.

Reply:
Yeah, and also `useDispatch`. I added the useEffect to fix an issue earlier, but it seems that's already fixed and it might not be needed. I think we still need the component to be able to dispatch the store function, or we need to wrap publicize.jsx into another HOC component